src/main/java/org/broadinstitute/hellbender/Main.java

```diff
@@ -98,6 +100,27 @@ protected static void printDecoratedExceptionMessage(final PrintStream ps, final
         return packageList;
     }
 
+
+    /**
+     * Reads from the given command-line arguments, pulls out configuration options,
+     * and initializes the configuration for this instance of Main.
+     *
+     * Suggested use for this is to handle downstream project configuration options and overrides.
+     * For example this would allow:
+     *
+     *      Custom command-line arguments for use in tools
+     *      Custom config file loading and initialization
+     *
+     * @param args The Array of command-line arguments.
+     */
+    protected static void parseArgsForConfigSetupAndGetNewArgs(final String[] args) {
+        // Now we setup our configurations:
+        // This method will modify the given argArrayList to remove the gatkConfigFileOption and value
+        // if they are specified.  This is so that the later validation of commandline arguments can still
+        // happen properly and so that we can have the configuration initialized as early as possible.
+        ConfigUtils.initializeConfigurationsFromCommandLineArgs(args, "--" + StandardArgumentDefinitions.GATK_CONFIG_FILE_OPTION);
```

Comment:
Comments to this method need updating, as the method no longer modifies the args array.